lib/jxl/decode.cc

```diff
@@ -1563,14 +1564,20 @@ JxlDecoderStatus JxlDecoderProcessCodestream(JxlDecoder* dec, const uint8_t* in,
           }
           dec->image_out_buffer_set = false;
 
+          bool has_ec = !dec->ib->extra_channels().empty();
           for (size_t i = 0; i < dec->extra_channel_output.size(); ++i) {
             void* buffer = dec->extra_channel_output[i].buffer;
             // buffer nullptr indicates this extra channel is not requested
             if (!buffer) continue;
+            if (!has_ec) {
+              JXL_WARNING(
+                  "Extra channels are not supported when callback is used");
```

Comment:
Doesn't chrome use the callback? Does that mean that alpha no longer works? Or is this only about extra channels beyond alpha?

Reply:
rgba works via callback. it only will fail if (float) callback is set *and* extra channels are requested.